lib/ansible/modules/cloud/azure/azure_rm_storageaccount.py

```diff
@@ -579,7 +579,7 @@ def update_account(self):
                 except Exception as exc:
                     self.fail("Failed to update tags: {0}".format(str(exc)))
 
-        if self.blob_cors and not compare_cors(self.account_dict.get('blob_cors', []), self.blob_cors):
+        if not compare_cors(self.account_dict.get('blob_cors', []), self.blob_cors):
```

Comment:
`if self.blob_cors` is used to prevent None value, but it also fails the empty list condition which causes the issue. Remove it here is not a good choice. `compare_cors` method will directly call `len(self.blob_cors)` and will throw error when `self.blob_cors` is None.

`if self.blob_cors is not None ...` should be more reasonable.

Reply:
@gavinfish Thanks for your review. You're right, I will update as your said!